src/idl_gen_rust.cpp

```diff
@@ -808,7 +808,7 @@ class RustGenerator : public BaseGenerator {
 
     // Get flatbuffers union key.
     // CASPER: add docstrings?
-    code_ += "  fn {{ENUM_NAME_SNAKE}}_type(&self) -> {{ENUM_NAME}} {";
+    code_ += "  pub fn {{ENUM_NAME_SNAKE}}_type(&self) -> {{ENUM_NAME}} {";
```

Comment:
This didn't seem deliberately private so I made it public instead of `pub(in crate::xyz_generated)`.

Reply:
Thanks!